test/upgrade/probe.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package upgrade
+
+import (
+	"context"
+	"flag"
+
+	pkgupgrade "knative.dev/pkg/test/upgrade"
+	"knative.dev/serving/test"
+	"knative.dev/serving/test/e2e"
+	v1test "knative.dev/serving/test/v1"
+)
+
+var successFraction = flag.Float64("probe.success_fraction", 1.0, "Fraction of probes required to pass during upgrade.")
```

Comment:
Maybe this needs to be configured? My suggestion would be to put it in one configuration struct and pass it via `kelseyhightower/envconfig`, see: https://github.com/knative/eventing/blob/241af67/test/upgrade/prober/configuration.go#L94-L119

Reply:
It's currently passed as a single flag. Why would we create a configuration struct for this? It sounds like you're proposing a different handling of parameters/arguments or environment and that should be tracked in a separate issue. This PR does not fix this.

Reply:
Fair enought